### PR TITLE
Fix for @4325--python2 testing not available on Github Actions.

### PR DESCRIPTION
**Pull request by kaby76**
This PR fixes https://github.com/antlr/antlr4/issues/4325.

This change removes the python2 target from Antlr. Github Actions removed support of any testing of Python 2.x on June 20, 2023. In addition, official support from python.org for 2.x stopped several years ago.

The instructions for the Python target have been updated with explicit code examples for a minimal parser driver, and the use of a visitor and listener to evaluate an expression of an expression grammar.

### CODE REVIEW

**Comment by ericvergnaud** [comment]
Yes I guess the next release would 4.14.
You also need to remove the test code, and the codegen code
Should be pretty straightforward

**Comment by kaby76** [comment]
Yes, looks easy.
```
grep . -R -i -e python2 | grep -v '[.]git'
grep . -R -i -e "python 2 | grep -v '[.]git'"
find . -iname 'python2'
```

File: doc/python-target.md [hunk 1]
```diff
@@ -2,17 +2,11 @@
 
 The examples from the ANTLR 4 book converted to Python are [here](https://github.com/jszheng/py3antlr4book).
 
-There are 2 Python targets: `Python2` and `Python3`. This is because there is only limited compatibility between those 2 versions of the language. Please refer to the [Python documentation](https://wiki.python.org/moin/Python2orPython3) for full details.
+There is ONE Python target: `Python3`. This is because there is only limited compatibility between those 2 versions of the language. Please refer to the [Python documentation](https://wiki.python.org/moin/Python3) for full details.
```
Comment (ericvergnaud) [inline doc/python-target.md:5]: I guess this can be rephrased a bit ?

File: doc/python-target.md [hunk 2]
```diff
@@ -21,9 +15,8 @@ For a full list of antlr4 tool options, please visit the tool documentation page
 
 ## Where can I get the runtime?
 
-Once you've generated the lexer and/or parser code, you need to download the runtime. The Python runtimes are available from PyPI:
+Once you've generated the lexer and/or parser code, you need to download the runtime. The Python runtime are available from PyPI:
```
Comment (ericvergnaud) [inline doc/python-target.md:18]: are -> is

**Comment by kaby76** [comment]
Thanks. I was about to Grammarly/ChatGPT the crap out of these files. I think I'll move the "Target Agnostic Format" into a completely different page. I now have some experience in getting target-agnostic working for all targets--probably more than anyone else.

**Comment by kaby76** [comment]
I rewrote the entire python-target.md file.
* The style of the English is more formal.
* Contractions are removed.
* Unclear antecedants are removed.
* Code snippets for hypothetical examples are replaced with a real-world example.

**Comment by kaby76** [comment]
https://github.com/antlr/antlr4/pull/4326/checks?check_run_id=14489678593

This is a bit of unnecessary overhead. The author name is already in the commit, and the author information already in the [/readme.md](https://github.com/antlr/antlr4/blob/96f97704a107db99d90ea9503d637104c3bb7afe/README.md?plain=1#L67), specifically, ***in every commit***.

How does this add to the quality of the product?

**Comment by ericvergnaud** [comment]
I agree that the DCO add-in isn't clever enough. If you can propose a better solution I will likely support it :-)

**Comment by kaby76** [comment]
According to the DCO code, there is a [isRequiredFor(author.login)](https://github.com/dcoapp/app/blob/de32b139d0f6c84670a764827a8878621913df53/lib/dco.js#L17C51-L17C51) test. Assuming auhor.login is not null, this check is returning "true", and goes to the next line where it fails [commit.verification.verified](https://github.com/dcoapp/app/blob/de32b139d0f6c84670a764827a8878621913df53/lib/dco.js#L18C41-L18C69), and gots to [the code that computes verification](https://github.com/dcoapp/app/blob/de32b139d0f6c84670a764827a8878621913df53/lib/dco.js#L33), and outputs [the message here](https://github.com/dcoapp/app/blob/de32b139d0f6c84670a764827a8878621913df53/lib/dco.js#L38). So, [getSignOffs](https://github.com/dcoapp/app/blob/de32b139d0f6c84670a764827a8878621913df53/lib/dco.js#L116) retyrbs a zero-length object. [Is it that you have to have the exact string "Signed-off-by:"??? ](https://github.com/dcoapp/app/blob/de32b139d0f6c84670a764827a8878621913df53/lib/dco.js#L122C19-L122C33) Who wrote this piece of poorly thought out crap?

**Comment by kaby76** [comment]
`-s -S` are git command options. Git in itself is warped because there is so much authentication already happening. I have been using Github Desktop for commits (I flip among different front ends because they all suck). But, Github Desktop doesn't have a way to specify the `-s -S` options. https://github.com/desktop/desktop/issues/5351 Blocked from using Github Desktop.

**Comment by ericvergnaud** [comment]
I use SourceTreeEnvoyé de mon iPhoneLe 23 juin 2023 à 13:54, Ken Domino ***@***.***> a écrit :﻿
-s -S are git command options. Git in itself is warped because there is so much authentication already happening. I have been using Github Desktop for commits (I flip among different front ends because they all suck). But, Github Desktop doesn't have a way to specify the -s -S options. desktop/desktop#5351 Blocked from using Github Desktop.

**Comment by ericvergnaud** [comment]
Let's move this DCO stuff to a conversation?

**Comment by kaby76** [comment]
> I use SourceTreeEnvoyé de mon iPhoneLe 23 juin 2023 à 13:54, Ken Domino ***@***.***> a écrit :﻿ -s -S are git command options. Git in itself is warped because there is so much authentication already happening. I have been using Github Desktop for commits (I flip among different front ends because they all suck). But, Github Desktop doesn't have a way to specify the -s -S options. [desktop/desktop#5351](https://github.com/desktop/desktop/issues/5351) Blocked from using Github Desktop. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

Yes, I also use SourceTree, but it also has issues, especially on Windows. They can't seem to get the UI working. But, there was something else, and I can't remember why I started using Github Desktop predominately for the last few months.

**Comment by kaby76** [comment]
Closing and reopening as another PR because `git rebase` does not work (conflicts), `git commit` doesn't work (insists on user name and password which does not work anymore with Github.com), etc., etc., etc.

**Pull request closed without merging**